txnkv/transaction/txn.go

```diff
@@ -343,6 +363,13 @@ func (txn *KVTxn) Commit(ctx context.Context) error {
 		sessionID = val.(uint64)
 	}
 
+	if txn.interceptor != nil {
+		// User has called txn.SetInterceptor() to explicitly set an interceptor, we
+		// need to bind it to ctx so that the internal client can perceive and execute
+		// it before initiating an RPC request.
+		ctx = tikvrpc.SetInterceptorIntoCtx(ctx, txn.interceptor)
```

Comment:
The "ctx = Set...(ctx, ...)" is annoying.

Reply:
Replaced with `WithRPCInterceptor`